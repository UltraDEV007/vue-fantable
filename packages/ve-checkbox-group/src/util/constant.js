// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
