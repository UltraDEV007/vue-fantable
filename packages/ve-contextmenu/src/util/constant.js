// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
