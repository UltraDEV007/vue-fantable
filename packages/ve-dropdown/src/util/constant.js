// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
