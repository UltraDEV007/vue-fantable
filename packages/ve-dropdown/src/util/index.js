/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
