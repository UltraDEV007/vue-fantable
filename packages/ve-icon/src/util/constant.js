// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
