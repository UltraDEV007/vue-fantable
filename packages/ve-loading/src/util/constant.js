// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
