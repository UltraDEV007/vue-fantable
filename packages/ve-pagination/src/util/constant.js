// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
