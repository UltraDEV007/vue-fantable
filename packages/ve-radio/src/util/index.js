import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
