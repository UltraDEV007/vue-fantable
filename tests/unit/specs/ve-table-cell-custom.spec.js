import { mount } from "@vue/test-utils";
import veTable from "@/ve-table";
import { later } from "../util";

describe("veTable cell custom", () => {
    const TABLE_DATA = [
        {
            name: "John",
            date: "1900-05-20",
            hobby: "coding",
            address: "No.1 Century Avenue, Shanghai",
        },
        {
            name: "Dickerson",
            date: "1910-06-20",
            hobby: "coding",
            address: "No.1 Century Avenue, Beijing",
        },
        {
            name: "Larsen",
            date: "2000-07-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Chongqing",
        },
        {
            name: "Geneva",
            date: "2010-08-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Xiamen",
        },
        {
            name: "Jami",
            date: "2020-09-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Shenzhen",
        },
    ];

    const FOOTER_DATA = [
        {
            rowkey: 0,
            name: "平均值",
            date: 213,
            hobby: 355,
            address: 189,
        },
        {
            rowkey: 1,
            name: "汇总值",
            date: 1780,
            hobby: 890,
            address: 2988,
        },
    ];

    const COLUMNS = [
        {
            field: "name",
            key: "a",
            title: "Name",
            align: "left",
            renderHeaderCell: ({ column }, h) => {
                return (
                    <span
                        class="header-cell-custom-class-test"
                        style="color:#1890ff;"
                    >
                        {column.name}
                    </span>
                );
            },
            renderBodyCell: ({ row, column, rowIndex }, h) => {
                return (
                    <span
                        class="body-cell-custom-class-test"
                        style="color:#1890ff;"
                    >
                        {++rowIndex}
                    </span>
                );
            },
            renderFooterCell: ({ row, column, rowIndex }, h) => {
                return (
                    <span class="footer-cell-custom-class-test" style="">
                        {row.name}
                    </span>
                );
            },
        },
        {
            field: "date",
            key: "b",
            title: "Date",
            align: "left",
        },
        {
            field: "hobby",
            key: "c",
            title: "Hobby",
            align: "left",
        },
        {
            field: "address",
            key: "d",
            title: "Address",
            width: "",
            align: "left",
        },
    ];

    const wrapper = mount(veTable, {
        propsData: {
            columns: COLUMNS,
            tableData: TABLE_DATA,
            footerData: FOOTER_DATA,
            rowKeyFieldName: "rowkey",
        },
    });

    it("render", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("cell custom", () => {
        // header
        expect(wrapper.find(".header-cell-custom-class-test").exists()).toBe(
            true,
        );

        // body
        expect(wrapper.find(".body-cell-custom-class-test").exists()).toBe(
            true,
        );

        // footer
        expect(wrapper.find(".footer-cell-custom-class-test").exists()).toBe(
            true,
        );
    });
});
