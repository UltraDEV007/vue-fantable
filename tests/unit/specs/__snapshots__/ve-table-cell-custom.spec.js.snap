// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FanTable cell custom render 1`] = `
<div class="vue-table-root">
  <div class="fan-table fan-table-border-around" style="width: 100%;">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col>
            <col>
            <col>
            <col>
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header">
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;"><span class="header-cell-custom-class-test" style="color: rgb(24, 144, 255);"></span></th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Date</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Hobby</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: left; top: 0px;">Address</th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;"><span class="body-cell-custom-class-test" style="color: rgb(24, 144, 255);">1</span></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">1900-05-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">coding</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shanghai</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;"><span class="body-cell-custom-class-test" style="color: rgb(24, 144, 255);">2</span></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">1910-06-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">coding</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Beijing</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;"><span class="body-cell-custom-class-test" style="color: rgb(24, 144, 255);">3</span></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2000-07-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Chongqing</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;"><span class="body-cell-custom-class-test" style="color: rgb(24, 144, 255);">4</span></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2010-08-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Xiamen</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;"><span class="body-cell-custom-class-test" style="color: rgb(24, 144, 255);">5</span></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2020-09-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shenzhen</td>
            </tr>
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer">
            <tr row-key="0" class="fan-table-footer-tr">
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: left; bottom: 0px;"><span class="footer-cell-custom-class-test">平均值</span></td>
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: left; bottom: 0px;">213</td>
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: left; bottom: 0px;">355</td>
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: left; bottom: 0px;">189</td>
            </tr>
            <tr row-key="1" class="fan-table-footer-tr">
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: left; bottom: 0px;"><span class="footer-cell-custom-class-test">汇总值</span></td>
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: left; bottom: 0px;">1780</td>
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: left; bottom: 0px;">890</td>
              <td rowspan="1" colspan="1" class="fan-table-footer-td" style="text-align: left; bottom: 0px;">2988</td>
            </tr>
          </tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left"></div>
          <div class="fan-table-selection-middle"></div>
          <div class="fan-table-selection-fixed-right"></div>
        </div>
      </div>
    </div>
    <div class="fan-table-edit-input-container" style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;"><textarea tabindex="-1" class="fan-table-edit-input"></textarea></div>
  </div>
</div>
`;
